src/backend/distributed/transaction/transaction_management.c

```diff
@@ -290,6 +290,13 @@ CoordinatedTransactionCallback(XactEvent event, void *arg)
 			dlist_init(&InProgressTransactions);
 			activeSetStmts = NULL;
 			CoordinatedTransactionUses2PC = false;
+
```

Comment:
@thanodnl I get into this assertion failure: https://github.com/citusdata/citus/issues/3535

Are you OK to remove the assert as Marco suggested?